modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py

```diff
@@ -2248,5 +2249,67 @@ def run_compare(df1, df2, **kwargs):
         )
 
 
+class TestDuplicateColumns:
+    def test_init(self):
+        def init(df, **kwargs):
+            return df
+
+        data = [
+            [1, 2, 3, 4],
+            [5, 6, 7, 8],
+            [9, 10, 11, 12],
+            [13, 14, 15, 16],
+            [17, 18, 19, 20],
+        ]
+        columns = ["c1", "c2", "c1", "c3"]
+        run_and_compare(
+            fn=init,
+            data=data,
+            force_lazy=False,
+            constructor_kwargs={"columns": columns},
+        )
+
+    def test_loc(self):
+        def loc(df, **kwargs):
+            df = df.loc[:, ["col1", "col3", "col3"]]
+            assert (
+                not isinstance(df, pd.DataFrame)
```

Comment:
Not sure why we need this assertion? The new `df` should definitely be an object `pd.DataFrame`.

Reply:
This method is called by run_and_compare() twice - one for pd.DataFrame and another one for pandas.DataFrame. The returned values are compared.

Reply:
Then this assertion will raise for pandas df because it doesn't have `._query_compiler._modin_frame._has_unsupported_data`. Or am I missing something?

Reply:
And why do we need to add this assertion only here, but not to add in other tests added?

Reply:
> Then this assertion will raise for pandas df because it doesn't have `._query_compiler._modin_frame._has_unsupported_data`. Or am I missing something?

No, it wont due to the isinstance check.



> And why do we need to add this assertion only here, but not to add in other tests added?

It was added following this https://github.com/modin-project/modin/pull/4896#discussion_r992042074 recommendation, but now I think that it's not only redundant, but incorrect, because, if support for duplicate columns added to the HDK backend, the test will fail. I think, I should remove this assertion.